### DISCUSSION TITLE
Bad detection mechanism (applies to all radios)

### DISCUSSION THREAD

**Discussion by galagithub** [hmatuschek/qdmr#420] [General]
I see that you use USB VID/PID to detect the cable for a specific radio, this is flawed!

For once, many radios have a USB to serial chip (like Prolific PL series or CH340, and so on) which makes detecting USB VID/PID combination useless as all the communication will be done across serial ports.

Twice, many cables with different USB-serial chips work with the same radio, I have 4 cables for Baofengs that have 4 different chipsets inside each with their own VID/PID, none of them matching what you hardcoded as radio+USB_VID/PID combination.

Please, either allow a set of USB VID/PID to be scanned, or add an settings window where we can specify the USB VID/PID.

**Comment by allesand** [comment]
@galagithub : so far, the way QDMR currently handles detection, worked for me on Anytone, Baofeng and Retevis as expected.
Are you using the cables that came in the box with the radios?

Since it works for most people, do you think calling someones work "flawed" gives the discussion a good direction?

**Comment by galagithub** [comment]
Yes, all the "official" cables that I've got use CH340 chip.

Do you think that assuming every other people on earth have exactly the same HW and SW setup as you is a good direction?

**Comment by allesand** [comment]
Re: #285 and #389, end of discussion for me.

**Comment by hmatuschek** [comment]
Thank you @galagithub for your valuable and measured suggestion on how QDMR should detect radios. Especially, as "just select the serial interface" nicely generalizes to other radios like Radioddity and TyT/Retevis devices, registering as DFU and HID USB devices. Being a software for several devices -- not just the one you happen own -- I have to unify all means to a single method to find and identify all supported radios. 

Of cause, having a single device/interface using different VID/PIDs makes my live hard, especially if AnyTone starts to change the VID/PID, althoug the interface remains the same. That latter I do not know for sure, maybe AnyTone changed the protocol and then, a new VID/PID is actually helpful.

**Comment by DH1PK** [comment]
Hi all!
I see the problem...

Maybe qdmr could offer the user to select vendor and model from two lists - and the, after the user picked the device type from the list, qdmr could test the available ports (usb/serial, hid, dfu,...) if the selected radio model responds?

Regards,
Patrick

**Reply by DH1PK** [reply]
Or maybe qdmr could probe first, and after not finding a radio it could offer a list to pick the model and port?

**Reply by hmatuschek** [reply]
There is something like that in the planning. In fact, I had once a branch implementing that.

**Reply by hmatuschek** [reply]
Can't find it any more. So #435.

**Comment by hmatuschek** [comment]
Hope that PR #436 does the trick for you.
